src/AppBundle/Repository/EventRepository.php

```diff
@@ -396,6 +397,83 @@ private function getUsernamesSql(Event $event): string
         return ltrim(implode(',', $usernames), ',');
     }
 
+    /**
+     * @param Event $event
+     * @param EventWiki $wiki
+     * @return int
+     */
+    public function getBytesChanged(Event $event, EventWiki $wiki): int
+    {
+        /** @var EventWikiRepository $eventWikiRepo */
+        $eventWikiRepo = $this->em->getRepository('Model:EventWiki');
+        $eventWikiRepo->setContainer($this->container);
+
+        $dbName = $eventWikiRepo->getDbNameFromDomain($wiki->getDomain());
+
+        $pageIds = $wiki->getPages();
+
+        if (!$pageIds) {
+            // IN () is an SQL syntax error
+            return 0;
+        }
+
+        return $this->getBytesInternal($event, $dbName, $pageIds);
+    }
+
+    /**
+     * @param Event $event
+     * @param string $dbName
+     * @param int[] $pageIds
+     * @return int
+     */
+    private function getBytesInternal(Event $event, string $dbName, array $pageIds): int
+    {
+        $revisionTable = $this->getTableName('revision');
+        $pageTable = $this->getTableName('page');
+        $userIds = $event->getParticipantIds();
+        $usernames = array_column($this->getUsernamesFromIds($userIds), 'user_name');
```

Comment:
This is already fetched and cached in EventProcessor. So you might pass it in to `getBytesChanged`. We should probably add query caching to Repository::getUsernamesFromIds() too, or instead... not sure what the ideal solution is.

Reply:
That would require moving the byte difference stuff to `setContributionsWikipedias` which already has a scary signature - do you think it's worth it?

Reply:
Meh, turned the parameters into fields.